### PR TITLE
Password Protected Files

### PR DESCRIPTION
**Pull request by dapocalypse**
This PR adds password protected files. This PR adds two new buttons to the file manager, encrypt, and decrypt. Apon pressing the encrypt button a prompt will appear, a password would then be typed into the prompt. Now after a password has been set whenever the user opens the file they are prompted with a password prompt. If they enter the previously set password the contents of the file will be shown. If they enter a different password the contents will not be shown. The decrypt button will remove the password requirement after a password check.

### CODE REVIEW

**Comment by Taaketa** [comment]
Feature freeze?

File: code/modules/modular_computers/file_system/computer_file.dm [hunk 1]
```diff
@@ -36,4 +37,16 @@ var/global/file_uid = 0
...
 	temp.filetype = filetype
-	return temp
\ No newline at end of file
+	temp.password = password
+	return temp
+
+/datum/computer_file/proc/can_access_file(var/mob/user, input_password = "")
+	if(!password)
+		return TRUE
+	else
```
Comment (Birdtalon) [inline code/modules/modular_computers/file_system/computer_file.dm]: Pur your else if statement on one line.

File: code/modules/modular_computers/file_system/programs/generic/file_browser.dm [hunk 2]
```diff
@@ -66,7 +67,8 @@
 						"name" = F.filename,
 						"type" = F.filetype,
 						"size" = F.size,
-						"undeletable" = F.undeletable
+						"undeletable" = F.undeletable,
+						"encrypted" = !!F.password
```
Comment (Birdtalon) [inline code/modules/modular_computers/file_system/programs/generic/file_browser.dm]: Not a fan of this kind of obfuscation.

File: code/modules/modular_computers/file_system/computer_file.dm [hunk 3]
```diff
@@ -36,4 +37,16 @@ var/global/file_uid = 0
...
+		if (!input_password)
```
Comment (Birdtalon) [inline code/modules/modular_computers/file_system/computer_file.dm]: We don't have spaces between the `if` keyword and the bracket for conditional statements in our style guide.

**Comment by Birdtalon** [comment]
You also have leading merge commits, which I didn't notice before.

File: code/modules/modular_computers/file_system/programs/generic/file_browser.dm (hunk 2 above)
Comment (tigercat2000) [reply code/modules/modular_computers/file_system/programs/generic/file_browser.dm]: This isn't obfuscation, it's boolean conversion. There's not really a more clean way to get the true/false representation of a variable in BYOND.

**Comment by dapocalypse** [comment]
Feature freeze?

File: code/modules/modular_computers/file_system/programs/generic/file_browser.dm (hunk 2 above)
Comment (Birdtalon) [reply code/modules/modular_computers/file_system/programs/generic/file_browser.dm]: And if you ask the majority of SS13 coders what this means do you think they'll know?

Comment (Birdtalon) [reply code/modules/modular_computers/file_system/programs/generic/file_browser.dm]: But you're right there isn't really a way in BYOND unfortunately.

**Comment by Crazylemon64** [comment]
#9426

File: code/modules/modular_computers/file_system/programs/generic/file_browser.dm (hunk 2 above)
Comment (KasparoVy) [reply code/modules/modular_computers/file_system/programs/generic/file_browser.dm]: Probably time for us to learn. *shrug*

**Comment by dapocalypse** [comment]
I fixed three bugs, can this be opened?

**Comment by dapocalypse** [comment]
#9656 #9635 #9610

**Pull request closed without merging**